Kernel/Clause.cpp

```diff
@@ -193,7 +193,8 @@ void Clause::destroy()
   static Stack<Clause*> toDestroy(32);
   Clause* cl = this;
   for(;;) {
-    if (env.options->proofExtra() == Options::ProofExtra::FULL) {
+    if ((env.options->proofExtra() == Options::ProofExtra::FULL) ||
+        ((env.options->questionAnswering() == Options::QuestionAnsweringMode::SYNTHESIS) && env.proofExtra.find(cl))) {
```

Comment:
I think `remove` should work also if the entry is not in the map, so you can probably remove the `find` call.